### PR TITLE
Types for Schema Id, Cred Def Id, Revoc Reg Id

### PR DESCRIPTION
**Pull request by Artemkaaas**

### CODE REVIEW

**Comment by AxelNennker** [comment]
Would you please consider to replace `ok_or(` by `ok_or_else(||`
Like here
`parts.get(0).ok_or(format!("Credential Definition Id validation failed: issuer DID not found in: {}", self.0))?;`
by
parts.get(0).ok_or_else(|| format!("Credential Definition Id validation failed: issuer DID not found in: {}", self.0))?;

File: libindy/src/domain/ledger/rev_reg_def.rs [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
+    pub cred_def_id: CredentialDefinitionId,
     pub value: RevocationRegistryDefinitionValue
 }
 
 impl RevRegDefOperation {
     pub fn new(rev_reg_def: RevocationRegistryDefinitionV1) -> RevRegDefOperation {
         RevRegDefOperation {
             _type: REVOC_REG_DEF.to_string(),
-            id: rev_reg_def.id.to_string(),
+            id: rev_reg_def.id.clone(),
```
Comment (jovfer) [inline libindy/src/domain/ledger/rev_reg_def.rs]: Is clone required here?

File: libindy/src/domain/ledger/rev_reg_def.rs [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
             type_: rev_reg_def.revoc_def_type.to_str().to_string(),
             tag: rev_reg_def.tag.to_string(),
-            cred_def_id: rev_reg_def.cred_def_id.to_string(),
+            cred_def_id: rev_reg_def.cred_def_id.clone(),
```
Comment (jovfer) [inline libindy/src/domain/ledger/rev_reg_def.rs]: Is clone required here?

File: libindy/src/commands/cache.rs [hunk 3]
```diff
@@ -222,7 +224,7 @@ impl CacheCommandExecutor {
                 LedgerCommand::GetCredDef(
                     pool_handle,
                     Some(submitter_did.to_string()),
-                    id.to_string(),
+                    id.clone(),
```
Comment (jovfer) [inline libindy/src/commands/cache.rs]: Is clone required here?

File: libindy/src/commands/anoncreds/prover.rs [hunk 4]
```diff
@@ -785,7 +785,7 @@ impl ProverCommandExecutor {
             attrs: credential_values,
             schema_id: credential.schema_id,
             cred_def_id: credential.cred_def_id,
-            rev_reg_id: credential.rev_reg_id.as_ref().map(|s| s.to_string()),
+            rev_reg_id: credential.rev_reg_id.clone(),
```
Comment (jovfer) [inline libindy/src/commands/anoncreds/prover.rs]: Is clone required here?

**Pull request merged**